src/singlelayerqg.jl

```diff
@@ -270,37 +276,63 @@ on the `grid`.
 """
 function set_zeta!(sol, vars, params, grid, zeta)
   mul!(vars.zetah, grid.rfftplan, zeta)
-  
+
   @. sol = vars.zetah
 
   updatevars!(sol, vars, params, grid)
-  
+
   return nothing
 end
 
 set_zeta!(prob, zeta) = set_zeta!(prob.sol, prob.vars, prob.params, prob.grid, zeta)
 
 
 """
-    energy(prob)
-    energy(sol, grid)
+    k_energy(prob)
+    p_energy(prob)
+    k_energy(vars, grid)
+    p_energy(vars, grid, params)
```

Comment:
Let's split the docstrings. We should have one for kinetic energy and one for potential energy.

Also, may I suggest the rather more verbose `kinetic_energy` and `potential_energy` for names of these functions?

(Let us not be haunted by ghosts of the past from the old-fortran times when variable names had to be restricted to a maximum of 8 or 12 characters... :))